pyproject.toml

```diff
@@ -85,6 +85,12 @@ dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
 [tool.ruff.per-file-ignores]
 "sympy/interactive/session.py" = ["F821"]
 
+[tool.ruff.lint]
+select=[
+    "SLOT000",
+    "SLOT001",
+    "SLOT002",
+]
```

Comment:
It is not so that ruff works. Read the ruff docs ;).
Moreover, install ruff locally on your machine, then you ca run the linting using the command 
```
python ruff check . --config pyproject.toml
```

From the working directory. 

In this way you don't have to push everytime you want to check if the config is working ;)

Reply:
Thanks for the help , its my first time working with it and I am thankful for your advice .

Reply:
```python
$ ruff check . --config pyproject.toml
warning: The top-level linter settings are deprecated in favour of their counterparts in the `lint` section. Please update the following options in `pyproject.toml`:
  - 'dummy-variable-rgx' -> 'lint.dummy-variable-rgx'
  - 'ignore' -> 'lint.ignore'
  - 'select' -> 'lint.select'
  - 'per-file-ignores' -> 'lint.per-file-ignores'
``` 

This is the warning i get , when i run it right now . I'll be looking more into the docs of ruff to figure out what i should do . I however fixed that came across my way . Kindly check them out once , I will try to fix the mistakes till then .

Reply:
@VascoSch92 I have tried fixing the problem in my latest PR , please take a look at it . I tried going through the documentation and made the changes . Kindly give me any more suggestions if needed .